site/src/request_handlers/github.rs

```diff
@@ -163,6 +160,68 @@ async fn handle_rust_timer(
     Ok(github::Response)
 }
 
+/// Parses the first occurrence of a `@rust-timer queue <...>` command
+/// in the input string.
+fn parse_queue_command(body: &str) -> Option<Result<QueueCommand, String>> {
+    let prefix = "@rust-timer";
+    let bot_line = body.lines().find_map(|line| {
+        let line = line.trim();
+        line.find(prefix)
+            .map(|index| line[index + prefix.len()..].trim())
+    })?;
+
+    let args = bot_line.strip_prefix("queue").map(|l| l.trim())?;
+    let mut args = match parse_command_arguments(args) {
+        Ok(args) => args,
+        Err(error) => return Some(Err(error)),
+    };
+    let mut cmd = QueueCommand {
+        include: args.remove("include"),
+        exclude: args.remove("exclude"),
+        runs: None,
+    };
+    if let Some(runs) = args.remove("runs") {
+        let Ok(runs) = runs.parse::<u32>() else {
+            return Some(Err(format!("Cannot parse runs {runs} as a number")));
+        };
+        cmd.runs = Some(runs as i32);
+    }
+
+    if let Some((key, _)) = args.into_iter().next() {
+        return Some(Err(format!("Unknown command argument {key}")));
+    }
```

Comment:
I guess you could just check that `args` is empty here, and dump all the remaining keys as unknown arguments in the error.